Rubberduck.Core/AutoComplete/Service/AutoCompleteService.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Rubberduck.Settings;
+using Rubberduck.VBEditor.Events;
+
+namespace Rubberduck.AutoComplete.Service
+{
+    public class AutoCompleteService : IDisposable
+    {
+        private readonly IGeneralConfigService _configService;
+        private readonly IEnumerable<AutoCompleteHandlerBase> _handlers;
+
+        private AutoCompleteSettings _settings;
+        private bool _popupShown;
+        private bool _enabled;
+        private bool _initialized;
+
+        public AutoCompleteService(IGeneralConfigService configService, IEnumerable<AutoCompleteHandlerBase> handlers)
+        {
+            _configService = configService;
+            _configService.SettingsChanged += ConfigServiceSettingsChanged;
+
+            _handlers = handlers;
+            InitializeConfig(); // todo: remove I/O work from constructor
+        }
+
+        private bool _initializing;
+
+        private void InitializeConfig()
+        {
+            _initializing = true;
+            // No reason to think this would throw, but if it does, _initializing state needs to be reset.
+            try
+            {
+                if (!_initialized)
+                {
+                    var config = _configService.LoadConfiguration();
+                    ApplyAutoCompleteSettings(config);
+                }
+            }
+            finally
+            {
+                _initializing = false;
+            }
+        }
+
+        public void Enable()
+        {
+            if (!_initializing)
```

Comment:
I assume that this should be `if (!_initialized)`. The initializing flag will usually be false when this is called, because it's reset before `InitializeConfig` returns.

Reply:
@Vogel612 I'd sorted that out in a prior commit - `_initializing` is there to prevent re-entry - `ApplyAutoCompleteSettings` can call `Enable`. `_initialized` is tested for prior to entry.  Which means... `Enable` and `Disable` should probably both be private.